esiclient/v1/node_network.py

```diff
@@ -264,21 +264,40 @@ def take_action(self, parsed_args):
         ironic_client = self.app.client_manager.baremetal
         neutron_client = self.app.client_manager.network
 
-        node = None
-        port = None
+        current_node = ironic_client.node.get(node_uuid)
 
-        with concurrent.futures.ThreadPoolExecutor() as executor:
-            f1 = executor.submit(ironic_client.node.get, node_uuid)
-            f2 = executor.submit(neutron_client.find_port, port_uuid)
-            node = f1.result()
-            port = f2.result()
-
-        if not port:
+        if current_node is None:
             raise exceptions.CommandError(
-                "ERROR: Port {1} not attached to node {0}".format(
-                    node.name, port_uuid))
+                "ERROR: The node {0} is not available".format(
+                    node_uuid))
+
+        if port_uuid:
+            port = neutron_client.find_port(port_uuid)
+            if not port:
+                raise exceptions.CommandError(
+                    "ERROR: Port {1} not attached to node {0}".format(
+                        current_node.name, port_uuid))
+        else:
+            ports = ironic_client.port.list(node=node_uuid, detail=True)
+
+            mapped_node_port_list = []
+            for port in ports:
+                if port.internal_info.get("tenant_vif_port_id"):
+                    mapped_node_port_list.append(port)
```

Comment:
I think you might as well append the neutron port here instead of the baremetal port; if you do, then line 296 becomes `port_uuid = mapped_port_list[0]`